exporters/tests/test_pelorus_webhook.py

```diff
@@ -0,0 +1,116 @@
+#
+# Copyright Red Hat
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+#
+
+
+import json
+import os
+from http import HTTPStatus
+
+import pytest
+from fastapi.testclient import TestClient
+
+from webhook import app as pelorus_webhook
+from webhook.plugins.pelorus_handler_base import PelorusWebhookPlugin
+
+client = TestClient(pelorus_webhook.app)
+
+test_directory = os.path.dirname(os.path.realpath(__file__))
```

Comment:
Lets use `pathlib` (also one of Python standard libraries), for handling paths

Reply:
Done. `os` vs `pathlib` is really not that big of a deal here. We don't use any functions that would be dependent on the operating system, so I don't see any advantage of pathlib in this situation.  Even more we need to convert pathlib objects to the strings, but if that's your preference I am also happy with it and changed to the pathlib module.

`os` is faster and have more native (unix like) feeling while using.

Reply:
see diff and decide:

```diff
diff --git a/exporters/tests/test_pelorus_webhook.py b/exporters/tests/test_pelorus_webhook.py
index 87fd4cd..63530e6 100644
--- a/exporters/tests/test_pelorus_webhook.py
+++ b/exporters/tests/test_pelorus_webhook.py
@@ -16,7 +16,7 @@
 
 
 import json
-import os
+from pathlib import Path
 from http import HTTPStatus
 
 import pytest
@@ -27,7 +27,7 @@ from webhook.plugins.pelorus_handler_base import PelorusWebhookPlugin
 
 client = TestClient(pelorus_webhook.app)
 
-test_directory = os.path.dirname(os.path.realpath(__file__))
+test_directory = Path(__file__).resolve().parent
 
 TEST_DATA_DIR = f"{test_directory}/data/"
 
diff --git a/exporters/webhook/app.py b/exporters/webhook/app.py
index 460ba32..d4fc1dc 100644
--- a/exporters/webhook/app.py
+++ b/exporters/webhook/app.py
@@ -19,7 +19,7 @@ import asyncio
 import http
 import importlib
 import logging
-import os
+from pathlib import Path
 import sys
 from typing import Optional, TypeVar
 
@@ -48,7 +48,7 @@ from webhook.store.in_memory_metric import (
 
 pelorus.setup_logging()
 
-script_directory = os.path.dirname(os.path.realpath(__file__))
+script_directory = Path(__file__).resolve().parent
 
 DEFAULT_PLUGIN_FOLDER = "plugins"
 
@@ -77,13 +77,14 @@ def register_plugin(webhook_plugin: PelorusWebhookPlugin):
 
 def load_plugins(plugins_dir_name: Optional[str] = DEFAULT_PLUGIN_FOLDER):
     plugin_path = f"{script_directory}/{plugins_dir_name}/"
-    sys.path.append(script_directory)
+    sys.path.append(f"{script_directory}")
     logging.info(f"Loading plugins from directory {plugin_path}")
-    if os.path.isdir(plugin_path):
-        for filename in os.listdir(plugin_path):
-            if filename.endswith("_handler.py"):
+    plugin_dir_path = Path(plugin_path)
+    if plugin_dir_path.is_dir():
+        for filename in plugin_dir_path.iterdir():
+            if filename.is_file() and filename.name.endswith("_handler.py"):
                 module = importlib.import_module(
-                    f".{filename[:-3]}", package=plugins_dir_name
+                    f".{filename.stem}", package=f"{plugins_dir_name}"
                 )
                 for name in dir(module):
                     obj = getattr(module, name)
```